### ISSUE TITLE
DBZ Legend of Z: 1 HP at beginning of battle despite event command requesting full heal before battle

### ISSUE THREAD

**Issue by OmegaMagnusDA** [EasyRPG/Player#1685]
Name of the game: **Dragon Ball Z: Legend of Z RPG**

Player platform: **Android** and possibly others

-Download D**BZ LOZ (No music version)** for quickest download (third option) https://gamejolt.com/games/dragon-ball-z-legend-of-z-rpg/145494

-**Save(s)**: https://drive.google.com/open?id=11MZSXtwmaNd-dBA-t488XCxxSf8qeIgG (enter house to the right and go through quick dialogue scenes to the battle, Goku vs Raditz at Roshi House)

-**easyrpg_log.txt log file**: https://drive.google.com/open?id=1O7Hr7P-puoby0-YWZQHLB1pKCR2AKyHJ

**Issue in detail**:
New error from 0.6. Before Goku engages in battle with Raditz, he has 1 HP even though there is a event command that requests full heal. The event commands on this map have not been changed before or after the release of 0.6. 
The commands are as listed below. It looks like the "-9999 HP" is being done before battle when its listed to be executed after battle. The 2nd picture are the battle events of the battle that are supposed to happen but since the party member is at 1 HP, gameover results when Raditz attacks.

Event Editor:
![image](https://user-images.githubusercontent.com/48193819/54091205-86c91500-433a-11e9-8af3-148af785c4fd.png)

Battle Events (from Monster Group/Troops):
![image](https://user-images.githubusercontent.com/48193819/54091782-5c2e8a80-4341-11e9-8d6a-4ae3ce1220d6.png)

**Comment by fmatthew5876** [comment]
Wow this one is super interesting.

If you do this code:
```
Battle
Change HP -9999
```

It works correctly

But if you do
```
Message: Hi
Battle
Change HP -9999
```

The battle starts with 1 HP!

So something about the the message really messes things up!

**Comment by fmatthew5876** [comment]
This is event more broken. If you change the above event to autostart, it just keeps showing "Hi" message box over and over, never starting the battle...

**Comment by fmatthew5876** [comment]
@CherryDT

Maybe you could help us with a little insight on how game message event commands work?

When a message is started what chunks get set?
What is the behavior for blocking other events execution exactly?
Any frame delays?
Any other wierd edge cases?

I can't really test this one with my save method unfortunately. Only before the message appears and after it's gone.

**Comment by fmatthew5876** [comment]
Test Cases 2 - Message box animations
---------------------------------------------

1. parallel event `Message: Hello`

Result: Open, "hello", Close, Open, "hello", etc..

2. parallel event `Loop: Message: Hello`

Result: Open, "hello", Close, Open, "hello", etc..

3. autostart event `Message: Hello`

Result: Open, "hello", "hello", "hello", etc..

4. autostart event `Loop: Message: Hello`

Result: Open, "hello", "hello", "hello", etc..

5. autostart event 
```
Loop: 
  Message: Hello
  Wait 0.0s (x4)
```

Result: Open, "hello", Closes halfway, Opens from halfway position, "hello", ...

Conclusion:
--------------

* Messages from autostart events don't wait for message box animations to complete.
* Message box shink and grow animations can be "interupted" and reversed before they complete.

**Comment by fmatthew5876** [comment]
Test Cases 4 - Message box timings
-----------------------------------

1. Set this parallel event
```
OpenSaveMenu
Msg: \^
OpenSaveMenu
```

| Save | stop count |
| -- | -- |
| 1 | 0 |
| 2 | 18 |

Conclusion:
---------------

Message box animations surely take 8 frames. As this example shows 8 + 8 + 2 frames used.

**Comment by fmatthew5876** [comment]
DBZ issue fixed by #1705

Proper message test cases and refactoring will be done later.
